addons/html_builder/static/src/core/save_plugin.js

```diff
@@ -57,29 +61,53 @@ export class SavePlugin extends Plugin {
         for (const getDirtyEls of this.getResource("get_dirty_els")) {
             dirtyEls.push(...getDirtyEls());
         }
-        const saveProms = dirtyEls.map(async (dirtyEl) => {
-            dirtyEl.classList.remove("o_dirty");
-            const cleanedEl = dirtyEl.cloneNode(true);
-            this.dispatchTo("clean_for_save_handlers", { root: cleanedEl });
+        // Group elements to save if possible
+        const groupedElements = groupBy(dirtyEls, (dirtyEl) => {
+            const model = dirtyEl.dataset.oeModel;
+            const field = dirtyEl.dataset.oeField;
 
-            if (this.config.isTranslation) {
-                await this.saveTranslationElement(cleanedEl);
-            } else {
-                const proms = this.getResource("save_element_handlers")
-                    .map((h) => h(cleanedEl))
-                    .filter(Boolean);
-                if (!proms.length) {
-                    console.warning("no save_element_handlers for dirty element", cleanedEl);
-                }
-                await Promise.all(proms);
+            // There are elements which have no linked model as something
+            // special is to be done "to save them". In that case, do not group
+            // those elements.
+            if (!model) {
+                return uniqueId("special-element-to-save-");
             }
+
+            const groupElHandler = this.getResource("group_element_id_handlers")[0];
+            // If not defined, group elements which are from the same field of
+            // the same record.
+            return groupElHandler(model, field) || `${model}::${dirtyEl.dataset.oeId}::${field}`;
+        });
+        const saveProms = Object.values(groupedElements).map(async (dirtyEls) => {
+            const cleanedEls = dirtyEls.map((dirtyEl) => {
+                dirtyEl.classList.remove("o_dirty");
+                const cleanedEl = dirtyEl.cloneNode(true);
+                this.dispatchTo("clean_for_save_handlers", { root: cleanedEl });
+                return cleanedEl;
+            });
+            const saveElementsHandler = this.getResource("save_elements_handlers")[0];
```

Comment:
Maybe a better name than `save_elements_handlers` (too close to the other `save_element_handlers`)

Reply:
Maybe check if we can remove one of them